### PR TITLE
Add flag -disable-poll-insertion

### PR DESCRIPTION
**Pull request by gretay-js**
Disables insertion of poll points.

### CODE REVIEW

File: driver/flambda_backend_args.ml [hunk 1]
```diff
@@ -59,6 +59,9 @@ let mk_alloc_check f =
 let mk_dcheckmach f =
   "-dcheckmach", Arg.Unit f, " (undocumented)"
 
+let mk_disable_poll_insertion f =
+  "-disable-poll-insertion", Arg.Unit f, " do not insert poll points"
```
Comment (xclerc) [inline driver/flambda_backend_args.ml:63]:
```suggestion
  "-disable-poll-insertion", Arg.Unit f, " Do not insert poll points"
```

File: backend/polling.ml [hunk 2]
```diff
@@ -261,7 +261,8 @@ let find_poll_alloc_or_calls instr =
   List.rev !matches
 
 let instrument_fundecl ~future_funcnames:_ (f : Mach.fundecl) : Mach.fundecl =
-  if function_is_assumed_to_never_poll f.fun_name then f
+  if !Flambda_backend_flags.disable_poll_insertion ||
+     function_is_assumed_to_never_poll f.fun_name then f
```
Comment (xclerc) [inline backend/polling.ml:265]: There is one instance of `function_is_assumed_to_never_poll`
(line 153, outside of the diff) that is not being augmented
with the additional condition. Is that on purpose?

File: backend/polling.ml [hunk 3]
```diff
@@ -278,7 +279,8 @@ let instrument_fundecl ~future_funcnames:_ (f : Mach.fundecl) : Mach.fundecl =
   end
 
 let requires_prologue_poll ~future_funcnames ~fun_name i =
-  if function_is_assumed_to_never_poll fun_name then false
+  if !Flambda_backend_flags.disable_poll_insertion ||
```
Comment (xclerc) [inline backend/polling.ml:282]: (Maybe go through a utility function.)

Comment (mshinwell) [reply backend/polling.ml:282]: This seems ok I think, I'd like to get this in soon...

**Review by mshinwell — APPROVED** [review]

**Review by xclerc — APPROVED** [review]

**Pull request merged**